### PR TITLE
Bump jsonrpc-core to v15

### PR DESCRIPTION
**Pull request by maciejhirsz**
Depends on https://github.com/paritytech/substrate/pull/7157

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2414,7 +2414,7 @@ checksum = "ecbdaacc17243168d9d1fa6b2bd7556a27e1e60a621d8a2a6e590ae2b145d158"
 dependencies = [
  "failure",
  "futures 0.1.29",
- "jsonrpc-core",
+ "jsonrpc-core 14.2.0",
```
Comment (gnunicorn) [inline Cargo.lock:2417]: Why does this still contain `14.2.0` versions? We updated all jsonrpc to `15.0`, right? By here some dependencies are specified as still bound to `jsonrpc-core 14.2.0`, is that intentional?

**Review by gnunicorn — APPROVED** [review]
But some version mixing I'd like to see explained.

File: Cargo.lock (hunk 1 above)
Comment (maciejhirsz) [reply Cargo.lock:2417]: Ah, I commited this after building with current substrate master, that's why it's there.

**Comment by gnunicorn** [comment]
bot merge

**Review by ordian — APPROVED** [review]

**Pull request merged**